subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestExecutionBuildConfigurationAction.java

```diff
@@ -102,6 +103,27 @@ public void execute(JavaDebugOptions javaDebugOptions) {
                 testTasksToRun.add(testTask);
             }
         }
+
+        for (InternalTestPatternSpec patternSpec : testExecutionRequest.getTestPatternSpecs()) {
+            for (Test task : queryTestTasks(patternSpec.getTaskPath())) {
```

Comment:
Should this be `Test` or `AbstractTestTask`?

Reply:
Yes, fixed.